docs/src/main/sphinx/security/https.rst

```diff
@@ -0,0 +1,361 @@
+:ref:`JKS <glossJKS>` keystore containers, as well as those imported into the
+binary :ref:`PKCS #12 <glossPKCS12>` archive file format.
+
+Make sure you obtain a certificate that is validated by a recognized
+:ref:`certificate authority <glossCA>`.
+
+Inspect received certificates
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Be sure to inspect and validate your received key and certificate files to make
+sure they reference the correct information to access your Trino server. Much
+unnecessary debugging time is saved by taking the time to validate your
+certificates before proceeding to configure the server.
+
+Inspect PEM-encoded files as described in :doc:`Inpsect PEM files
+</security/inspect-pem>`.
+
+Inspect JKS keystores as described in :doc:`Inspect JKS files
+</security/inspect-jks>`.
+
+Invalid certificates
+^^^^^^^^^^^^^^^^^^^^^
+
+If your certificate does not pass validation, or does not show the expected
+information on inspection, contact the group or vendor who provided it for a
+replacement.
+
+.. _cert-placement:
+
+Place the certificate file
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+There are no location requirements for a certificate file as long as:
+
+* The location is visible from the server's configuration file location with a
+  relative path reference from the server's root directory, or with an absolute
+  path reference.
+* The location is secure from copying or tampering by malicious actors.
+
+You can place your file in the Trino server's ``etc`` directory, which allows
+you to use a relative path reference in configuration files. However, this
+location can require you to keep track of the certificate file, and move it to
+a new ``etc`` directory when you upgrade your Trino version.
+
+.. _configure-https:
+
+Configure the coordinator
+^^^^^^^^^^^^^^^^^^^^^^^^^
+
+On the coordinator, add the following lines to the :ref:`config properties file
+<config_properties>` to enable HTTPS support for the server.
+
+.. note::
+
+  Legacy ``keystore`` and ``truststore`` wording is used in property names, even
+  when directly using PEM-encoded certificates. PEM certs generally do not
+  require truststore properties, because this format incorporates its chain of
+  authorizing certificates back to the originating CA (or possibly back to an
+  intermediate CA assigned by a large international CA).
+
+.. code-block:: text
+
+  http-server.https.enabled=true
+  http-server.https.port=8443
+  http-server.https.keystore.path=etc/clustercoord.pem
+
+Possible alternatives for the third line include:
+
+.. code-block:: text
+
+  http-server.https.keystore.path=etc/clustercoord.jks
+  http-server.https.keystore.path=/usr/local/certs/clustercoord.p12
+
+Relative paths are relative to the Trino server's root directory. In a
+``tar.gz`` installation, the root directory is one level above ``etc``.
+
+JKS keystores always require a password, while PEM format certificates can
+optionally require a password. For these cases, add the following line to
```

Comment:
`For the cases where you need a password, add the following line to the configuration`?

Reply:
OK